docs/src/test/scala/docs/http/scaladsl/HttpServerWithTypedSpec.scala

```diff
@@ -0,0 +1,187 @@
+    import akka.actor.typed.ActorSystem
+    import akka.util.Timeout
+
+    import akka.http.scaladsl.server.Directives._
+    import akka.http.scaladsl.model.StatusCodes
+    import akka.http.scaladsl.server.Route
+    import akka.http.scaladsl.server.directives.MethodDirectives.delete
+    import akka.http.scaladsl.server.directives.MethodDirectives.post
+    import akka.http.scaladsl.server.directives.RouteDirectives.complete
+    import akka.http.scaladsl.server.directives.PathDirectives.path
+
+    import scala.concurrent.duration._
+    import scala.concurrent.Future
+    import BuildJobRepository._
+
+    class JobRoutes(system: ActorSystem[_], buildJobRepository: ActorRef[BuildJobProtocol]) extends JsonSupport {
+
+      import akka.actor.typed.scaladsl.AskPattern._
+
+      // asking someone requires a timeout and a scheduler, if the timeout hits without response
+      // the ask is failed with a TimeoutException
+      implicit val timeout: Timeout = 3.seconds
+      implicit val scheduler = system.scheduler
+
+      lazy val theJobRoutes: Route =
+        pathPrefix("jobs") {
+          concat(
+            pathEnd {
+              concat(
+                post {
+                  entity(as[Job]) { job =>
+                    val operationPerformed: Future[Response] = buildJobRepository.ask(replyTo => AddJob(job, replyTo))
+                    onSuccess(operationPerformed) {
+                      case OK         => complete("Job added")
+                      case KO(reason) => complete(StatusCodes.InternalServerError -> reason)
+                    }
+                  }
+                },
+                delete {
+                  val operationPerformed: Future[Response] = buildJobRepository.ask(replyTo => ClearJobs(replyTo))
+                  onSuccess(operationPerformed) {
+                    case OK         => complete("Jobs cleared")
+                    case KO(reason) => complete(StatusCodes.InternalServerError -> reason)
+                  }
+                }
+              )
+            },
+            (get & path(LongNumber)) { id =>
+              val maybeJob: Future[Option[Job]] = buildJobRepository.ask(replyTo => GetJobById(id, replyTo))
+              rejectEmptyResponse {
+                complete(maybeJob)
+              }
+            }
+          )
+        }
+    }
+    //#akka-typed-route
+
+    //#akka-typed-bootstrap
+    import akka.{ actor, Done }
+    import akka.actor.typed.scaladsl.adapter._
+    import akka.stream.ActorMaterializer
+
+    import akka.http.scaladsl.Http
+
+    import scala.concurrent.ExecutionContextExecutor
+    import scala.util.{ Success, Failure }
+
+    val system = ActorSystem[Done](Behaviors.setup[Done] { ctx =>
+      // http doesn't know about akka typed so create untyped system/materializer
+      implicit val untypedSystem: actor.ActorSystem = ctx.system.toUntyped
+      implicit val materializer: ActorMaterializer = ActorMaterializer()(ctx.system.toUntyped)
```

Comment:
There actually is an `akka-stream-typed` module with a `akka.stream.typed.scaladsl.ActorMaterializer` for this. Note entirely sure if that should go in the sample or not though.